provider/pkg/provider/ids.go

```diff
@@ -53,14 +93,24 @@ func getDefaultName(
 		randLength = autoNamingSpec.MinLength - len(prefix)
 	}
 
+	// Generate naming trivia for the resource.
+	namingTrivia := createNamingTrivia(sdkName, news)
+
 	maxLength := 0
 	if autoNamingSpec.MaxLength > 0 {
-		left := autoNamingSpec.MaxLength - len(prefix)
+		left := autoNamingSpec.MaxLength - len(prefix) - len(namingTrivia.Suffix)
```

Comment:
We have `autoNammingSpec` that contains metadata about auto-naming. My idea would be to add a `Suffix` property to it. The flow would go like this:
- Naming trivia (similar to yours) are defined in the `schema` folder, not `provider`
- Codegen uses that trivia to build `autoNamingSpec` metadata including Min/MaxLength and suffix
- Metadata are written down to `metadata.json`
- Engineers can easily inspect which resources and properties were affected by looking at the JSON diff
- Provider uses the metadata to build names at runtime, without knowing about `namingTrivia`

This flow is used in Azure Native (and AWS Native for min/max length).

What do you think?